### PR TITLE
Add support for get-poetry.py --file using sdist

### PR DESCRIPTION
**Pull request by DustinMoriarty**
# Pull Request Check List

Resolves: #3228

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

* This pull request adds support for install from an sdist file for the get-poetry script. Currently, the file structure of sdist vs the github releases is different due to the poetry package being nested one level deeper in the sdist file structure. Strictly speaking, this is a feature. However, the previous behavior may have resulted in a user experience that was difficult to distinguish from a bug.
* Check the library location and raise an error if the expected poetry library is not created from the file provided by the user.
* Added some unit tests for the new functionality in the get-poetry.py file. The test fixtures used and the import structure can be used for more unit tests of get-poetry.py in the future.

### CODE REVIEW

File: tests/scripts/test_get_poetry.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import os
+import shutil
+
+
+try:
+    # Python 3.5+
```
Comment (abn) [inline tests/scripts/test_get_poetry.py:6]: You can drop compatibility changes as we no longer test under py2 or py35.

File: tests/scripts/test_get_poetry.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import pytest
+
+from poetry.core.masonry.builder import Builder
+from poetry.factory import Factory
+
+
+# We need to set the poetry home environment variable prior to importing the script since it gets set once at import
+# time.
+SOURCE_HOME = Path(__file__).parent.parent.parent
```
Comment (abn) [inline tests/scripts/test_get_poetry.py:26]: Use the fixture. And convert the following to fixtures as well.

File: tests/scripts/test_get_poetry.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+GET_POETRY_PATH = SOURCE_HOME / "get-poetry.py"
+TMP_DIR = Path(__file__).parent / "tmp"
```
Comment (abn) [inline tests/scripts/test_get_poetry.py:28]: Use the existing`tmp_dir` fixture

**Review by abn — COMMENTED** [review]
I'm not sure why this is required because`sdists` are not supported in this install mechanism. And really, shouldn't be supported via `get-poetry.py`.

**Comment by DustinMoriarty** [comment]
I was wondering if that would be the case. Perhaps improved docs and error handling is the way to go. Thoughts?

**Comment by DustinMoriarty** [comment]
Also, just curious what the story is with all the OS specific builds. From what I can tell they all just have the poetry source in them. However, I am guessing there is an important use case that I am missing.

**Comment by abn** [comment]
The github releases contain vendored packages under `poetry/_vendored`. These could be unfortunately `(interpreter, os)` specific.

Improving the help doc for `--file`to indicate that only github releases are supported. As for error handling I guess you could validate the archive in-memory before extracting.

File: tests/scripts/test_get_poetry.py (hunk 1 above)
Comment (DustinMoriarty) [reply tests/scripts/test_get_poetry.py:6]: Well that simplifies things!

**Comment by DustinMoriarty** [comment]
I am going to close this pull request because sdist files will not be supported. See PR #3259 for an alternative resolution to issue #3228 by improving documentation and source file validation.

**Pull request closed without merging**